torch/fx/passes/infra/partitioner.py

```diff
@@ -132,6 +172,27 @@ def merge_single_node(node: Node, id: Optional[int]):
             elif id not in partitions_by_id:
                 assignment[node] = id
                 partitions_by_id[id] = Partition(id=id, nodes=[node])
+
+                # Create an entry in partition map for this partition id and iterate
+                # over all the nodes that are encountered in a dfs path of this node.
+                # We then add the partition id of all the nodes that are encountered
+                # in the dfs path of this node to indicate that there is a path from
+                # partiton id of node to partition id of curr_node.
+                dfs_path = dfs_paths_from_node[node]
+                for curr_node in dfs_path:
+                    target_id = assignment.get(curr_node, None)
+                    if target_id is not None:
+                        partition_map[id].add(target_id)
+
+                # Get the list of all the nodes that are in the dfs path of this node
+                # when starting from the input side of the node. Iterate through each node
+                # in this list and update the partition map to indicate that there is a path
+                # from the partition id of curr_node to the target partition id.
+                dfs_path = dfs_paths_to_node[node]
+                for curr_node in dfs_path:
+                    source_id = assignment.get(curr_node, None)
+                    if target_id is not None:
+                        partition_map[source_id].add(id)
             else:
                 assignment[node] = id
                 partitions_by_id[id].add_node(node)
```

Comment:
Do you also need to update partition_map in this branch? 

IIUC, whenever a node assignment is done, you would need to update partition_map ?

Reply:
Good catch, i need to address the case where partition id is updated for a single node. Let me fix that.

Reply:
Fixed this in latest version.